### PR TITLE
Improve search toolbar UX a little bit

### LINKED ISSUE #8095: Search input doesn't get focused when searching extensions/library
### Steps to reproduce

1. Go to the Browse screen, select the Extensions tab.
2. Click on the Search button on the toolbar.

### Expected behavior

After clicking the search button, the search input should show and gets focused, with the keyboard appearing.

### Actual behavior

The input doesn't get focused, and the keyboard doesn't appear, you need a second tap on the input for it to show.

### Crash logs

_No response_

### Tachiyomi version

r4977

### Android version

Android 12

### Device

Samsung Galaxy S20+

### Other details

Probably introduced by the Compose rewrite.

It also happens with the library search.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by alessandrojean**
Changes:
- Change search toolbar text field style to `titleMedium` with `fontSize = 18.sp` to match Stable.
- Add support to `placeholderText` in the search toolbar text field.
- Set IME action as search and clear the focus and hide the keyboard when needed after tapping IME search.

| Before | After |
| -- | -- |
|  |  |
|  |  |

### CODE REVIEW

File: app/src/main/java/eu/kanade/presentation/browse/SourceSearchScreen.kt [hunk 1]
```diff
@@ -10,6 +10,8 @@ import androidx.paging.compose.collectAsLazyPagingItems
 import eu.kanade.domain.manga.model.Manga
 import eu.kanade.presentation.browse.components.BrowseSourceSearchToolbar
 import eu.kanade.presentation.components.Scaffold
+import eu.kanade.tachiyomi.R
+import eu.kanade.tachiyomi.glance.stringResource
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/presentation/browse/SourceSearchScreen.kt:14]: This is the wrong `stringResource`

Comment (alessandrojean) [reply app/src/main/java/eu/kanade/presentation/browse/SourceSearchScreen.kt:14]: Thought I had fixed that, probably forgot to add to the commit, thanks for the heads-up.

**Comment by alessandrojean** [comment]
Reverted the `FocusRequest` fix in favour of #8098.

**Comment by arkon** [comment]
There's a merge conflict.

**Comment by alessandrojean** [comment]
Conflict fixed 👍🏻

**Pull request merged**